cgp/genome.py

```diff
@@ -90,6 +90,9 @@ def __init__(
         # dictionary to store values of Parameter nodes
         self._parameter_names_to_values: Dict[str, float] = {}
 
+        # list of permissible values for every gene
+        self._permissible_values = self.determine_permissible_values()
```

Comment:
```suggestion
        self._permissible_values: List[np.ndarray] = self.determine_permissible_values()
```

Reply:
could you also further specify the type, e.g., `List[np.ndarray[int]]`?

Reply:
I think that `np.ndarray[int]` is not a valid type hint. There is the `nptyping` package which supposedly achieves such functionality but I think this is a bit of an overkill here.